app/models/P5/departure/ConsignmentItem.scala

```diff
@@ -16,63 +16,96 @@
+  val dangerousGoods: String                   = Commodity.DangerousGoods.map(_.showAll).getOrElse("")
+  val cusCode: String                          = Commodity.cusCode.getOrElse("")
+  val descriptionOfGoods: String               = Commodity.descriptionOfGoods
+  val previousDocumentString: String           = PreviousDocument.map(_.showAll).getOrElse("")
+  val supportingDocumentString: String         = SupportingDocument.map(_.showAll).getOrElse("")
+  val transportDocumentString: String          = TransportDocument.map(_.showAll).getOrElse("")
+  val additionalReferenceString: String        = AdditionalReference.map(_.showAll).getOrElse("")
+  val additionalInformationString: String      = AdditionalInformation.map(_.showAll).getOrElse("")
 
   val grossMass: String          = Commodity.GoodsMeasure.grossMass.toString
   val netMass: String            = Commodity.GoodsMeasure.netMass.getOrElse("").toString
-  val consignorId: String        = Consignor.flatMap(_.identificationNumber).getOrElse("")
   val consigneeId: String        = Consignee.flatMap(_.identificationNumber).getOrElse("")
   val supplyChainActorId: String = AdditionalSupplyChainActor.map(_.map(_.identificationNumber).mkString("; ")).getOrElse("")
 
-  val totalPackages: Int = Packaging.foldLeft(0)(
-    (total, packaging) => total + packaging.numberOfPackages.getOrElse(0)
-  )
-
   val packagesType: String = Packaging.map(_.toString).mkString("; ")
+  val totalPackages        = Packaging.length
+
+  val allProducedDocuments: Seq[ProducedDocument] =
+    (for {
+      supportingDocs <- SupportingDocument
+      transportDocs  <- TransportDocument
+    } yield {
+      val supportingProducedDocs = supportingDocs.map {
+        supportingDocument =>
+          ProducedDocument(
+            DocumentType(supportingDocument.`type`.getOrElse(""), "", transportDocument = false),
+            supportingDocument.referenceNumber,
+            supportingDocument.complementOfInformation
+          )
+      }
+      val transportProducedDocs = transportDocs.map {
+        transportDocument =>
+          ProducedDocument(DocumentType(transportDocument.`type`.getOrElse(""), "", transportDocument = true), transportDocument.referenceNumber, None)
+      }
+
+      supportingProducedDocs ++ transportProducedDocs
+    }).getOrElse(Seq.empty)
+
+  val packagingFormat: NonEmptyList[viewmodels.Package] =
+    NonEmptyList
+      .fromList(Packaging.flatMap {
+        case PackagingP5(_, Some(numberOfPackages), typeOfPackages, Some(shippingMarks)) =>
+          Some(RegularPackage(KindOfPackage(typeOfPackages, ""), numberOfPackages, shippingMarks))
+        case PackagingP5(_, None, typeOfPackages, shippingMarks) =>
+          Some(BulkPackage(KindOfPackage(typeOfPackages, ""), shippingMarks))
+        case PackagingP5(_, _, typeOfPackages, shippingMarks) =>
+          Some(UnpackedPackage(KindOfPackage(typeOfPackages, ""), 0, shippingMarks))
+        case _ => None
+      }.toList)
+      .get
```

Comment:
are we sure the get will not fail? Should we provide a default? Or maybe ConsignmentItem.Packaging should be a NonEmptyList too.